plugins/Alias/config.py

```diff
@@ -43,5 +43,7 @@ def configure(advanced):
 Alias = conf.registerPlugin('Alias')
 conf.registerGroup(Alias, 'aliases')
 conf.registerGroup(Alias, 'escapedaliases')
+conf.registerGlobalValue(Alias, 'validName',
+    registry.String(r'^[^\x00-\x20]+$', _("""Regex which alias names must match in order to be valid""")))
```

Comment:
Pretty much all characters seem to work fine, with the exception of whitespace (disallowed by registry), and characters stripped by callbacks.canonicalName. The only reason I can see to exclude characters other than whitespace and callbacks.canonicalName stripped characters is that the bot owner does not want them in alias names. I'll add an additional non-configurable check for whitespace.
(You could keep it as a whitelist by not merging d9b21dc, which was deliberately placed in a separate commit).

I'm not sure how making this configurable allows bot owners to break their bot + config file. Can you provide an example?